systemd/crc-wait-node-ready.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+set -o pipefail
+set -o errexit
+set -o nounset
+set -o errtrace
+
+source /usr/local/bin/crc-systemd-common.sh
+
+MAX_RETRY=150
+WAIT_SEC=2
+NODE_NAME=node/crc
+# Loop from 1 up to max_retry
+for retry in $(seq 1 "$MAX_RETRY"); do
+    node_status=$(oc get "$NODE_NAME" --no-headers | awk '{print $2}' || true)
+
+    # Check if the node status is "Ready"
+    if [[ $node_status == "Ready" ]]; then
+        echo "CRC node is ready."
+        exit 0
+    fi
+
+    echo "CRC node is not ready. Status: $node_status"
+
+    # If it's the last attempt, log a failure message before exiting
+    if (( retry == MAX_RETRY )); then
+        echo "Error: Timed out waiting for the CRC node to be ready after $MAX_RETRY attempts x $WAIT_SEC seconds." >&2
+        exit 1
+    fi
+
```

Comment:
Why this commit required? we are already doing lots of check around apiservice so if node is not ready those checks might not even work?